### PR TITLE
Weather app features

### PR DESCRIPTION
**Pull request by daviidy**
- I wrote the functions that hit the API. Functions that can take a location and return the weather data for that location.
- Set up a simple form that will let users input their location and fetch the weather info
- Styled the page using bootstrap 4

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @daviidy 👋

**Good job here so far. You have clearly covered some aspects. 👌**

However, let's make some fixes;

- Please refer to the comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: dist/index.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            <h1 class="display-1 font-weight-bold text-white">Weather App</h1>
+          </div>
+        </div>
+        <div class="row">
+          <div class="col-12 col-md-12 text-center search">
+            <form id="form">
+              <input type="text" name="search" value="" placeholder="Search for a city" id="searchInput"><br>
+              <button class="bg-blue pointer" type="submit" name="button" id="search">Search</button>
+            </form>
+            <p id="loading" class="text-white" style="display: none;">
```
Comment (mikenath223) [inline dist/index.html]: - [x] Please remove the inline styling here. Consider moving it to the stylesheet.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+const humidity = document.getElementById('humidity');
+const alert = document.getElementById('alert');
+const danger = document.getElementById('danger');
+const icon = document.getElementById('icon');
+const deg = document.getElementById('deg');
+const fahr = document.getElementById('fahr');
+const details1 = document.getElementById('details1');
+const details2 = document.getElementById('details2');
+const img = document.createElement('img');
+
```
Comment (mikenath223) [inline src/index.js]: - [x] Kindly separate the dom-manipulation aspect from the data fetching logic.

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      danger.style.display = 'block';
+      danger.innerHTML = 'Sorry we could not find any city for your search';
+      search.style.display = 'block';
+      loading.style.display = 'none';
+      setTimeout(() => {
+        danger.style.display = 'none';
+      }, 5000);
+    }
+  } catch (e) {
+    alert(`There is an error${e}`);
```
Comment (mikenath223) [inline src/index.js]: - [ ] Please use better error handling here. Alerts slow code running.

File: package.json [hunk 4]
```diff
@@ -9,7 +9,20 @@
...
   "license": "ISC",
   "devDependencies": {
+    "css-loader": "^5.0.2",
+    "style-loader": "^2.0.0",
     "webpack": "^5.21.2",
     "webpack-cli": "^4.5.0"
+  },
+  "dependencies": {
+    "bootstrap": "^4.6.0",
+    "eslint": "^7.19.0",
```
Comment (mikenath223) [inline package.json:22]: - [x] Kindly add the necessary start scripts in package.json:
```
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```

Comment (mikenath223) [reply package.json:22]: - If you would be including the start script section in the script then you would need to add the webpack-dev-server package also.

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @daviidy, great job with this project :+1: The DOM manipulations task has been separated from the application logic :100: The issue with the error handling has also been resolved :fireworks: Awesome. The necessary scripts have also been implemented. Great job :smile: 
However, there are a few issues we need to address before proceeding. Kindly check the comments below:
- The issue with the "inline styling" is not addressed fully. We still have inline styles in our "index.html" file
- The appearance should change based on the data received. Currently, this functionality is not addressed

[Optional]
- The "start" script returns an error when I run "npm run start". The webpack-dev-server is not set up correctly


Once these are resolved, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by bellom — APPROVED** [review]
## Hi @daviidy  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**